R/bm-tpc-h.R

```diff
@@ -639,6 +639,83 @@ tpc_h_queries[[10]] <- function(input_func) {
     collect()
 }
 
+tpc_h_queries[[11]] <- function(input_func) {
+  # SELECT
+  #     ps_partkey,
+  #     sum(ps_supplycost * ps_availqty) AS value
+  # FROM
+  #     partsupp,
+  #     supplier,
+  #     nation
+  # WHERE
+  #     ps_suppkey = s_suppkey
+  #     AND s_nationkey = n_nationkey
+  #     AND n_name = 'GERMANY'
+
+  partsupp <- input_func("partsupp")
+  supplier <- input_func("supplier")
+  nation <- input_func("nation")
+
+  joined_filtered <- partsupp %>%
+    inner_join(supplier, by = c("ps_suppkey" = "s_suppkey")) %>%
+    inner_join(nation, by = c("s_nationkey" = "n_nationkey")) %>%
+    filter(n_name == "GERMANY")
+
+  # GROUP BY
+  #     ps_partkey
+  # HAVING
+  #     sum(ps_supplycost * ps_availqty) > (
+  #         SELECT
+  #             sum(ps_supplycost * ps_availqty) * 0.0001000000
+  #         FROM
+  #             partsupp,
+  #             supplier,
+  #             nation
+  #         WHERE
+  #             ps_suppkey = s_suppkey
+  #             AND s_nationkey = n_nationkey
+  #             AND n_name = 'GERMANY')
+  # ORDER BY
+  #     value DESC;
+
+  # grouped mutate/filter works in dplyr but errors in Arrow because
+  # window functions aren't currently supported
+  # result1 <- joined_filtered %>%
+  #   mutate(global_value = sum(ps_supplycost * ps_availqty) * 0.0001000000) %>%
+  #   group_by(ps_partkey) %>%
+  #   mutate(value = sum(ps_supplycost * ps_availqty)) %>%
+  #   filter(value > global_value) %>%
+  #   ungroup() %>%
+  #   arrange(desc(value)) %>%
+  #   select(-global_value) %>%
+  #   collect()
+
+  # ...but not in Arrow yet, so we do a summarize() + left_join()
+  global_agr <- joined_filtered %>%
+    mutate(global_agr_key = 1L) %>%
+    group_by(global_agr_key) %>%
+    summarise(
+      global_value = sum(ps_supplycost * ps_availqty) * 0.0001000000
+    )
+
+  partkey_agr <- joined_filtered %>%
+    group_by(ps_partkey) %>%
+    summarise(value = sum(ps_supplycost * ps_availqty))
+
+  result2 <- joined_filtered %>%
+    mutate(global_agr_key = 1L) %>%
+    left_join(global_agr, by = "global_agr_key") %>%
+    left_join(partkey_agr, by = "ps_partkey") %>%
+    filter(value > global_value) %>%
+    arrange(desc(value)) %>%
+    select(-global_agr_key, -global_value) %>%
```

Comment:
```suggestion
    select(ps_partkey, value) %>%
```

I've been using the positive selection like this since it's in the query itself. I agree that  `select(-global_agr_key, -global_value)` is more likely for what someone would do in R if they were writing this, but the queries themselves do have which columns will be selected, so I think we can go with that.

Reply:
Also oddly(?) when I pulled this branch locally the `value` column didn't seem to show up in the error messages I got. I didn't dig in to why, but it's totally possible something is funky (in Arrow's hash join implementation) with the join / that name. Or possibly it's a bug in how we do negative selection? That might be worth investigating (separately from this, and can be deferred!)